File: fsharp/AoC2021/Day05/Day05/Program.fs
open System
open System.IO

// https://adventofcode.com/2021/day/5

let args = Environment.GetCommandLineArgs()

if args.Length <> 2 then
    failwith $"Usage: {args[0]} <INPUT FILE>"

let lines = File.ReadLines(args[1]) |> List.ofSeq |> List.map Solution.Line.parse

printfn $"Not considering diagonals, there are {Solution.countDangerousPoints false lines} points where multiple lines overlap."
printfn $"Considering diagonals, there are {Solution.countDangerousPoints true lines} points where multiple lines overlap."

File: csharp/AoC2021/Day05/Day05/Program.cs
// https://adventofcode.com/2021/day/5

using Day05;

if (args.Length != 1)
{
    Console.Error.WriteLine("Usage: $0 <input file>");
    Environment.Exit(1);
}

var lines = File.ReadAllLines(args[0]).Select(Line.Parse).ToList();

var count1 = Solution.CountDangerousPoints(lines, false);
Console.WriteLine($"Not considering diagonals, there are {count1} points where multiple lines overlap.");

var count2 = Solution.CountDangerousPoints(lines, true);
Console.WriteLine($"Considering diagonals, there are {count2} points where multiple lines overlap.");
